crates/egui_kittest/src/renderer.rs

```diff
@@ -0,0 +1,81 @@
+use egui::{Context, FullOutput, TexturesDelta};
+use image::RgbaImage;
+
+pub trait TestRenderer {
+    /// We use this to pass the glow / wgpu render state to [`eframe::Frame`].
+    #[cfg(feature = "eframe")]
+    fn setup_eframe(&self, _cc: &mut eframe::CreationContext<'_>, _frame: &mut eframe::Frame) {}
+
+    /// Handle a [`TexturesDelta`] by updating the renderer's textures.
+    fn handle_delta(&mut self, delta: &TexturesDelta);
+
+    /// Render the [`crate::Harness`] and return the resulting image.
+    ///
+    /// # Errors
+    /// Returns an error if the rendering fails.
+    fn render(&mut self, ctx: &Context, output: &FullOutput) -> Result<RgbaImage, String>;
+}
+
+/// A lazy renderer that initializes the renderer on the first render call.
```

Comment:
Would be _even_ better if we wouldn't need to initialize the renderer for each TestHarness individually :thinking:
(But that might get hairy quickly and a user can always bring their own renderer for that :))